tests/models/test_popular.py

```diff
@@ -212,3 +219,146 @@ def test_i2i(
         model = PopularModel()
         assert_second_fit_refits_model(model, dataset)
+
+
+class TestPopularModelConfiguration:
+    @pytest.mark.parametrize("begin_from", (None, datetime(2021, 11, 23), "2021-11-23T10:20:30.400+02:30"))
+    @pytest.mark.parametrize(
+        "period",
+        (
+            None,
+            timedelta(days=7),
+            {
+                "days": 7,
+                "seconds": 123,
+                "microseconds": 12345,
+                "milliseconds": 32,
+                "minutes": 2,
+                "weeks": 7,
+            },
+        ),
+    )
+    def test_from_config(
+        self, period: tp.Optional[tp.Union[timedelta, dict]], begin_from: tp.Optional[tp.Union[datetime, str]]
+    ) -> None:
+        config = {
+            "popularity": "n_interactions",
+            "period": period,
+            "begin_from": begin_from,
+            "add_cold": True,
+            "inverse": True,
+            "verbose": 0,
+        }
+        if period is not None and begin_from is not None:
+            with pytest.raises(ValueError):
+                model = PopularModel.from_config(config)
+        else:
+            model = PopularModel.from_config(config)
+            assert model.popularity.value == "n_interactions"
+            serialized_period = timedelta(**period) if isinstance(period, dict) else period
+            assert model.period == serialized_period
+            serialiazed_begin_from = datetime.fromisoformat(begin_from) if isinstance(begin_from, str) else begin_from
+            assert model.begin_from == serialiazed_begin_from
+            assert model.add_cold is True
+            assert model.inverse is True
+            assert model.verbose == 0
+
+    @pytest.mark.parametrize("begin_from", (None, datetime(2021, 11, 23)))
+    @pytest.mark.parametrize(
+        "period",
+        (
+            None,
+            timedelta(days=7),
+        ),
+    )
+    @pytest.mark.parametrize("popularity", ("mean_weight", "sum_weight"))
+    def test_get_config(
+        self,
+        popularity: tp.Literal["n_users", "n_interactions", "mean_weight", "sum_weight"],
+        period: tp.Optional[timedelta],
+        begin_from: tp.Optional[datetime],
+    ) -> None:
+        if period is not None and begin_from is not None:
+            with pytest.raises(ValueError):
+                model = PopularModel(
+                    popularity=popularity,
+                    period=period,
+                    begin_from=begin_from,
+                    add_cold=False,
+                    inverse=False,
+                    verbose=1,
+                )
+        else:
+            model = PopularModel(
+                popularity=popularity,
+                period=period,
+                begin_from=begin_from,
+                add_cold=False,
+                inverse=False,
+                verbose=1,
+            )
+            config = model.get_config()
+            serialized_period = (
```

Comment:
too difficult, let's better but this expected value manually in `parametrize` block

Reply:
Fixed.